azurerm/data_source_redis_cache.go

```diff
@@ -0,0 +1,266 @@
+package azurerm
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmRedisCache() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmRedisCacheRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"location": locationForDataSourceSchema(),
+
+			"resource_group_name": resourceGroupNameForDataSourceSchema(),
+
+			"zones": singleZonesSchema(),
+
+			"capacity": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"family": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"minimum_tls_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"shard_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+
+			"enable_non_ssl_port": {
+				Type:     schema.TypeBool,
+				Optional: true,
```

Comment:
since this is readonly - can we make this Computed instead of Optional?

```suggestion
				Computed: true,
```